### ISSUE TITLE
Incorrect type check with inherited class

### ISSUE THREAD

**Issue by PaleNeutron** [microsoft/pyright#10681]
**Describe the bug**
function parameter type is correct at parent class but incorrect from child class

**Code or Screenshots**
If possible, provide a minimal, self-contained code sample (surrounded by triple back ticks) to demonstrate the issue. The code should define or import all referenced symbols.

```python
from pydantic import ConfigDict


class SQLModelConfig(ConfigDict):
    table: bool
    registry: int

cd = ConfigDict(from_attributes=True, use_attribute_docstrings=True)


scd = SQLModelConfig(from_attributes=True, use_attribute_docstrings=True)
```

<img width="914" height="313" alt="Image" src="https://github.com/user-attachments/assets/2214526f-08e3-4373-b0f0-597814478705" />

**VS Code extension or command-line**
Are you running pyright as a VS Code extension, a language server in another editor, integrated into Pylance, or the command-line tool? Which version?

Use it integrated with pylanc, from settings `python.analysis.typeCheckingMode`

<img width="811" height="475" alt="Image" src="https://github.com/user-attachments/assets/8a25c7e5-b0df-49ca-9a83-cc47ef6df410" />

pylance version: 2025.6.2

**Comment by erictraut** [comment]
Pyright is working correctly here. The behavior in question is dictated by the Python typing spec, and pyright is conformant with the spec. For details, refer to [this chapter](https://typing.python.org/en/latest/spec/typeddict.html#typeddict).

The subclass is adding two new items with keys `table` and `registry`. By default, TypedDict items are required, which means that you must supply values for these two items when calling the `SQLModelConfig` constructor. If your intent is for these items to be "not required", you can either specify `total=False` in the class definition or use a `NotRequired` type qualifier for each of the new fields.

```python
class SQLModelConfig(ConfigDict, total=False):
    table: bool
    registry: int
```

or

```python
class SQLModelConfig(ConfigDict):
    table: NotRequired[bool]
    registry: NotRequired[int]
```
